modules/programs/firefox.nix

```diff
@@ -244,13 +266,15 @@ in
         package =
           if versionAtLeast config.home.stateVersion "19.09"
           then cfg.package.override { cfg = fcfg; }
```

Comment:
Does this line work on Darwin? If it does then I would suggest removing the below change and instead prompt Darwin users to use `stateVersion ≥ 19.09`.

Reply:
As far as I know, the wrapper doesn't play nice on darwin. The Firefox package has been broken on darwin for quite some time.

Reply:
Right, sorry I got myself a bit confused. But then this would cause issues for Darwin users having `stateVersion ≥ 19.09`, right? Since they'd end up in this `then` branch. Perhaps need something like

```
package =
  if isDarwin then
    cfg.package
  else if versionAtLeast config.home.stateVersion "19.09" then
    cfg.package.override { cfg = fcfg; }
  else
    (pkgs.wrapFirefox.override { config = bcfg; }) cfg.package { };
```

it also might be good to add `visible = !isDarwin;` to the `enableAdobeFlash`, `enableGoogleTalk`, and `enableIcedTea` options since they can't be used if `isDarwin`. Then these options shouldn't show up in the manual.

Reply:
Ahhh, good catch 👍 I wasn't thinking about that clearly.  
Nice, I didn't know about `visible`, I'll make these changes now :)

Reply:
Alright, I've pushed some more changes in 8f8003a, what d'you think?

Reply:
Ah, hold on 🙄 I didn't verify properly - building the manpage failed, so I just need to fix that!

Reply:
Since I haven't used `visible` before, any ideas what these indicate?
```
building '/nix/store/rvyk4528nrhv6cdzpvs9662d8yl0hi3s-manual-combined.drv'...

/nix/store/2kk3vxwk8ljxj3p5iviqpdhping8p9xd-manual-combined/manual-combined.xml:8523: element xref: validity error : IDREF attribute linkend references an unknown ID "opt-programs.firefox.enableIcedTea"
  8519  <itemizedlist>
  8520  <listitem>
  8521  <simpara>
  8522  The <xref linkend="opt-programs.firefox.enableGoogleTalk"/> and
  8523    <xref linkend="opt-programs.firefox.enableIcedTea"/> options are now deprecated
  8524    and will only work if Firefox ESR 52.x is used.
  8525  </simpara>

/nix/store/2kk3vxwk8ljxj3p5iviqpdhping8p9xd-manual-combined/manual-combined.xml:8522: element xref: validity error : IDREF attribute linkend references an unknown ID "opt-programs.firefox.enableGoogleTalk"
  8518  <simpara>This release has the following notable changes:</simpara>
  8519  <itemizedlist>
  8520  <listitem>
  8521  <simpara>
  8522  The <xref linkend="opt-programs.firefox.enableGoogleTalk"/> and
  8523    <xref linkend="opt-programs.firefox.enableIcedTea"/> options are now deprecated
  8524    and will only work if Firefox ESR 52.x is used.

/nix/store/2kk3vxwk8ljxj3p5iviqpdhping8p9xd-manual-combined/manual-combined.xml fails to validate
builder for '/nix/store/rvyk4528nrhv6cdzpvs9662d8yl0hi3s-manual-combined.drv' failed with exit code 3
```

Reply:
Ah, the manual references these options so we can't remove their visibility. So leave out the `visible = !isDarwin;` line for now.

I'll sort it out some other time 🙂

Reply:
Ah okay, makes sense! There you go, I've just pushed that to  c8badb9